syntax = "proto3";

package ivoice;

service IVoiceToolkit {
  rpc transcribeAudioFile (TranscribeRequest) returns (stream TranscribeResponse) {}
  rpc extractKeywords (ResultContent) returns (ResultKeywords) {}
}

message ResultContent {
  string content = 1;
  int32 count = 2;
}

message ResultKeywords {
  string keywords = 1;
}

message Segment {
  float start = 1;
  float end = 2;
}

message TranscribeRequest {
  string remoteFilePath = 1;
}


message TranscribeResponse {
  Segment segment = 1;
  int32 label = 2;
  string word = 3;
}